packages/kit/types/config.d.ts

```diff
@@ -1,38 +1,83 @@
 import { UserConfig as ViteConfig } from 'vite';
 import { RecursiveRequired } from './helper';
-import { Logger, TrailingSlash } from './internal';
+import { HttpMethod, Logger, RouteSegment, TrailingSlash } from './internal';
 
-export interface AdapterUtils {
+export interface RouteDefinition {
+	type: 'page' | 'endpoint';
+	pattern: RegExp;
+	segments: RouteSegment[];
+	methods: HttpMethod[];
+}
+
+export interface AdapterEntry {
```

Comment:
it could help to comment this one. questions I have: what's an adapter entry (is there one or multiple), what's the ID represent, what does the filter do?

Reply:
I've added some docs to the best of my abilities — it's not the easiest API to describe (though I've searched in vain for ways to simplify it further)